examples/versatileab/src/bin/undef-exception.rs

```diff
@@ -0,0 +1,33 @@
+//! Example triggering an undefined exception.
+
+#![no_std]
+#![no_main]
+
+// pull in our start-up code
+use versatileab as _;
+
+use semihosting::println;
+
+versatileab::entry_point!();
+
+/// The main function of our Rust application.
+#[export_name = "main"]
+fn main() -> ! {
+    println!("Hello, this is an undefined exception example");
+    unsafe {
+        core::arch::asm!("udf #0");
+    }
+    unreachable!("should never be here!");
+}
+
+#[no_mangle]
+unsafe extern "C" fn _undefined_handler(_faulting_instruction: u32) {
+    static mut COUNTER: u32 = 0;
```

Comment:
Re-entering this function using a static mut is UB. It should probably be at AtomicU32 using fetch_add.